pysdsl/types/intvector.hpp

```diff
@@ -169,88 +182,111 @@ inline auto add_int_class(py::module& m, py::dict& dict, KEY_T key,
+                    m, int_vectors_dict, "dynamic", "IntVector", doc_int_vector)
+                .def(
+                    py::init([](size_t size,
+                                uint64_t default_value,
+                                uint8_t bit_width) {
+                        return sdsl::int_vector<0>(size, default_value, bit_width);
+                    }),
+                    py::arg("size") = 0,
+                    py::arg("default_value") = 0,
+                    py::arg("bit_width") = 64,
+                    py::call_guard<py::gil_scoped_release>())
+                .def(
+                    "expand_width",
+                    [](sdsl::int_vector<0> &self, size_t width) {
+                        sdsl::util::expand_width(self, width); },
+                    "Expands the integer width to new_width >= v.width().",
+                    py::call_guard<py::gil_scoped_release>())
+                .def("bit_compress",
+                    [](sdsl::int_vector<0> &self) {
+                        sdsl::util::bit_compress(self); },
+                    doc_bit_compress,
+                    py::call_guard<py::gil_scoped_release>());
+    }
+
+    auto operator()(std::integral_constant<size_t, 1>) {
+        return add_int_class<sdsl::int_vector<1>, bool>(
                 m, int_vectors_dict, 1ul , "BitVector")
             .def(py::init(
                 [](size_t size, bool default_value) {
                     return sdsl::int_vector<1>(size, default_value, 1); }),
                 py::arg("size") = 0, py::arg("default_value") = false)
             .def("flip", &sdsl::int_vector<1>::flip,
                  "Flip all bits of bit_vector",
-                 py::call_guard<py::gil_scoped_release>()),
+                 py::call_guard<py::gil_scoped_release>());
+    }
+};
 
-        add_int_class<sdsl::int_vector<4>, uint16_t>(
-                m, int_vectors_dict, 4, "Int4Vector")
-            .def(py::init(
-                [](size_t size, uint8_t default_value) {
-                    return sdsl::int_vector<4>(size, default_value, 4); }),
-                py::arg("size") = 0, py::arg("default_value") = 0),
 
-        add_int_class<sdsl::int_vector<8>, uint16_t>(
-                m, int_vectors_dict, 8, "Int8Vector")
-            .def(py::init(
-                [](size_t size, uint8_t default_value) {
-                    return sdsl::int_vector<8>(size, default_value, 8); }),
-                py::arg("size") = 0, py::arg("default_value") = 0),
+template <typename... Ts>
+struct SubsetFunctor {
+    const std::tuple<Ts...>& tpl;
 
-        add_int_class<sdsl::int_vector<16>, uint16_t>(
-                m, int_vectors_dict, 16, "Int16Vector")
-            .def(py::init(
-                [](size_t size, uint16_t default_value) {
-                    return sdsl::int_vector<16>(size, default_value, 16); }),
-                py::arg("size") = 0, py::arg("default_value") = 0),
+    // it doesn't work in C++14
```

Comment:
separate `template .. make_subset_functor()` to avoid upgrade to c++17 (for now)?